### ISSUE TITLE
  Building modules, stage 2.   MODPOST 1 modules WARNING: "__aeabi_uldivmod" [/home/xw/workspace/github/openwrt-master/build_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/linux-mvebu_cortexa9/ndpi-netfilter-a360566-2.6/ndpi-netfilter/src/xt_ndpi.ko] undefined!

### ISSUE THREAD

**Issue by panda-mute** [vel21ripn/nDPI#44]
32-bit CPU.

Sorry, I can't figure out which file cause this issue.

**Comment by panda-mute** [comment]
diff --git a/src/lib/third_party/src/lruc.c b/src/lib/third_party/src/lruc.c
index a63ed99..b450fde 100644
--- a/src/lib/third_party/src/lruc.c
+++ b/src/lib/third_party/src/lruc.c
@@ -156,10 +156,11 @@ lruc *lruc_new(uint64_t cache_size, uint32_t average_length) {
     perror("LRU Cache unable to create cache object");
     return NULL;
   }
-  cache->hash_table_size      = cache_size / average_length;
   cache->average_item_length  = average_length;
   cache->free_memory          = cache_size;
   cache->total_memory         = cache_size;
+  do_div(cache_size, average_length);
+  cache->hash_table_size      = cache_size;
 #ifndef __KERNEL__
   cache->seed                 = time(NULL);
 #else

**Comment by panda-mute** [comment]
```
diff --git a/src/lib/third_party/src/lruc.c b/src/lib/third_party/src/lruc.c
index a63ed99..b450fde 100644
--- a/src/lib/third_party/src/lruc.c
+++ b/src/lib/third_party/src/lruc.c
@@ -156,10 +156,11 @@ lruc *lruc_new(uint64_t cache_size, uint32_t average_length) {
     perror("LRU Cache unable to create cache object");
     return NULL;
   }
-  cache->hash_table_size      = cache_size / average_length;
   cache->average_item_length  = average_length;
   cache->free_memory          = cache_size;
   cache->total_memory         = cache_size;
+  do_div(cache_size, average_length);
+  cache->hash_table_size      = cache_size;
 #ifndef __KERNEL__
   cache->seed                 = time(NULL);
 #else
```

**Comment by vel21ripn** [comment]
Thanks! Fixed in 23e3cbd40f19cc691840ec0ba3edf3a506499045

**Comment by panda-mute** [comment]
Got it, thanks.
